include/SDL3/SDL_gpu.h

```diff
@@ -44,49 +44,174 @@ extern "C" {
+ * after SDL_EndGPUComputePass is called.
+ *
+ * \since This struct is available since SDL 3.0.0
+ */
 typedef struct SDL_GPUComputePass SDL_GPUComputePass;
+
+/**
+ * An opaque handle representing a copy pass.
+ * This handle is transient and should not be held or referenced
+ * after SDL_EndGPUCopyPass is called.
+ *
+ * \since This struct is available since SDL 3.0.0
+ */
 typedef struct SDL_GPUCopyPass SDL_GPUCopyPass;
+
+/**
+ * An opaque handle representing a fence.
+ *
+ * \since This struct is available since SDL 3.0.0
+ */
 typedef struct SDL_GPUFence SDL_GPUFence;
 
+/**
+ * Specifies the primitive topology of a graphics pipeline.
+ *
+ * \since This enum is available since SDL 3.0.0
+ */
 typedef enum SDL_GPUPrimitiveType
 {
-    SDL_GPU_PRIMITIVETYPE_POINTLIST,
-    SDL_GPU_PRIMITIVETYPE_LINELIST,
-    SDL_GPU_PRIMITIVETYPE_LINESTRIP,
-    SDL_GPU_PRIMITIVETYPE_TRIANGLELIST,
-    SDL_GPU_PRIMITIVETYPE_TRIANGLESTRIP
+    SDL_GPU_PRIMITIVETYPE_POINTLIST,     /**< A series of separate points. */
+    SDL_GPU_PRIMITIVETYPE_LINELIST,      /**< A series of separate lines. */
+    SDL_GPU_PRIMITIVETYPE_LINESTRIP,     /**< A series of connected lines. */
+    SDL_GPU_PRIMITIVETYPE_TRIANGLELIST,  /**< A series of separate triangles. */
+    SDL_GPU_PRIMITIVETYPE_TRIANGLESTRIP  /**< A series of connected triangles. */
 } SDL_GPUPrimitiveType;
 
+/**
+ * Specifies how the contents of a texture attached to a render pass
+ * are treated at the beginning of the render pass.
+ *
+ * \since This enum is available since SDL 3.0.0
+ */
 typedef enum SDL_GPULoadOp
 {
-    SDL_GPU_LOADOP_LOAD,
-    SDL_GPU_LOADOP_CLEAR,
-    SDL_GPU_LOADOP_DONT_CARE
+    SDL_GPU_LOADOP_LOAD,      /**< The previous contents of the texture will be preserved. */
+    SDL_GPU_LOADOP_CLEAR,     /**< The contents of the texture will be cleared to a color. */
+    SDL_GPU_LOADOP_DONT_CARE  /**< The previous contents of the texture need not be preserved. The contents will be undefined. */
 } SDL_GPULoadOp;
 
+/**
+ * Specifies how the contents of a texture attached to a render pass
+ * are treated at the end of the render pass.
+ *
+ * \since This enum is available since SDL 3.0.0
+ */
 typedef enum SDL_GPUStoreOp
 {
-    SDL_GPU_STOREOP_STORE,
-    SDL_GPU_STOREOP_DONT_CARE
+    SDL_GPU_STOREOP_STORE,     /**< The contents generated during the render pass will be written to memory. */
+    SDL_GPU_STOREOP_DONT_CARE  /**< The contents generated during the render area are not needed and may be discarded. The contents will be undefined. */
 } SDL_GPUStoreOp;
 
+/**
+ * Specifies the size of elements in an index buffer.
+ *
+ * \since This enum is available since SDL 3.0.0
+ */
 typedef enum SDL_GPUIndexElementSize
 {
-    SDL_GPU_INDEXELEMENTSIZE_16BIT,
-    SDL_GPU_INDEXELEMENTSIZE_32BIT
+    SDL_GPU_INDEXELEMENTSIZE_16BIT, /**< The index elements are 16-bit. */
+    SDL_GPU_INDEXELEMENTSIZE_32BIT  /**< The index elements are 32-bit. */
```

Comment:
Silly question... why is this an enum and not a byte size?

Reply:
This is an enum on every backend API. Values other than 16 and 32 don't make sense.